internal/provider/resource_consul_cluster.go

```diff
@@ -92,9 +93,17 @@ func resourceConsulCluster() *schema.Resource {
 				Optional:         true,
 				ValidateDiagFunc: validateSemVer,
 				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
-					// Suppress diff is normalized versions match OR min_consul_version is removed from the resource
-					// since min_consul_version is required in order to upgrade the cluster to a new Consul version.
-					return input.NormalizeVersion(old) == input.NormalizeVersion(new) || new == ""
+					// Suppress diff for non specified value
+					if new == "" {
+						return true
+					}
+
+					actualConsulVersion := version.Must(version.NewVersion(old))
+					currentTFVersion := version.Must(version.NewVersion(new))
+					log.Printf("[DEBUG] Actual Consul Version %v", old)
+					log.Printf("[DEBUG] Current TF Version %v", new)
+					// suppres diff if the specified min_consul_version is <= to the actual consul version
+					return currentTFVersion.LessThanOrEqual(actualConsulVersion)
```

Comment:
Hmm, I entered a version less than my cluster's 1.9.6 version, but the diff does not seem to be suppressed 🤔 

resource "hcp_consul_cluster" "mycluster" {
  hvn_id          = hcp_hvn.example.hvn_id
  cluster_id      = "consul-cluster-tf"
  tier            = "development"
  public_endpoint = true
  min_consul_version = "v1.9.3"
}

```
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # hcp_consul_cluster.mycluster will be updated in-place
  ~ resource "hcp_consul_cluster" "mycluster" {
        id                            = "/project/2a385cdb-189b-4a05-8bc4-6b91c62fd703/hashicorp.consul.cluster/consul-cluster-tf"
      + min_consul_version            = "v1.9.3"
        # (23 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?```

Reply:
Nvm, I did not switch to this branch before building the provider 🤦 
All good.

Reply:
I think we identified together that was because the tf build wasn't refreshed.